sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getApplicationProgressDetails.e2e-spec.ts

```diff
@@ -0,0 +1,129 @@
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+        createSecondDisbursement: true,
+      },
+    );
+    const [firstDisbursement, secondDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    // Create approved student appeal submitted yesterday.
+    const approvedAppealRequest = createFakeStudentAppealRequest();
+    const approvedAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [approvedAppealRequest],
+    });
+    await db.studentAppeal.save(approvedAppeal);
+    // Create declined student appeal submitted today.
+    const declinedAppealRequest = createFakeStudentAppealRequest();
+    declinedAppealRequest.appealStatus = StudentAppealStatus.Declined;
+    const declinedAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [declinedAppealRequest],
+    });
+    await db.studentAppeal.save(declinedAppeal);
+    // Create pending student appeal submitted today.
+    const pendingAppealRequest = createFakeStudentAppealRequest();
+    pendingAppealRequest.appealStatus = StudentAppealStatus.Pending;
+    const pendingAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [pendingAppealRequest],
+    });
+    await db.studentAppeal.save(pendingAppeal);
+    // Create declined application offering change request.
+    await saveFakeApplicationOfferingRequestChange(
+      db,
+      {
+        application,
+      },
+      {
+        initialValues: {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.DeclinedBySABC,
+        },
+      },
+    );
+    // Create approved application offering change request.
+    await saveFakeApplicationOfferingRequestChange(
+      db,
+      {
+        application,
+      },
+      {
+        initialValues: {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.Approved,
+        },
+      },
+    );
+    // Create pending application offering change request.
+    await saveFakeApplicationOfferingRequestChange(db, { application });
+    const endpoint = `/students/application/${application.id}/progress-details`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        applicationStatus: application.applicationStatus,
+        applicationStatusUpdatedOn:
+          application.applicationStatusUpdatedOn.toISOString(),
+        pirStatus: application.pirStatus,
+        firstCOEStatus: firstDisbursement?.coeStatus,
+        secondCOEStatus: secondDisbursement?.coeStatus,
```

Comment:
Minor. Null check is not required here.

Reply:
Won't it be a good idea to keep the check?

Reply:
Updated the code. Thanks @dheepak-aot